content/blog/2020-11-16-reproducibility-in-ml-my-tools/index.en.md

```diff
@@ -0,0 +1,122 @@
+
+It seemed nice: all I add to do was create a "meaningful" filename, such as `param_1_param_2_..._param_100.txt` plus a timestamp and dump all the results. I was so naive: at one point, I've got more than 100 files for that project only. Now, try finding which experiment was the best. 
+
+The problem with this approach is that it is complicated to manage all those files. Indeed, it is nearly impossible to be efficient when comparing all the results since that every time you create a new file, you need to go through all your results again. Also, sharing this kind of work between team members is pure insanity.
+
+Possible solutions for that specific problem should (1) efficiently allow a user to compare results between different runs or parameters, (2) easily save the results and (3) not add too much overhead to use. Before giving any good solution, let's discuss the second problem since the proposed solution also solves this problem.
+
+## Managing Experimentations
+I've been struck with the problem of managing my experimentation during my thesis work. I had no clear strategy other than writing the configuration of my experiment in my results file title. At first, it seemed "right," but at one point, I had more than 15 parameters to "log" in the filename. The names were so painful to read (especially at 2 in the morning before a meeting with my supervisor). I was amazed how fast things can get out of hand with what first seems like a quick win. The big problem with this kind of approach is that the length grows as quickly as the number of parameters. For sure, I could create a directory and sub-directory related to the parameters. However, the problem is still there. I will now have a tree of directories of many nodes and will need to swim in a pool of directories.
+
+## What Can Be a Good Solution for Managing Results and Experimentations?
+We need to solve both the problem of managing results and experimentations is a database where to log our results and the settings of any experiments. But, I'm pretty sure you are not interested in creating your own database. Fortunately for us, different solutions exist to track results and experimentations. I will not go into details for all of them, but only focus on the two useful for my research projects. 
+
+### MLflow
+[MLflow](https://mlflow.org/docs/latest/index.html) is an open-source platform for tracking experiments to record, compare parameters and results. Using MLflow, you will get a visual interface (GUI) (Figure 1) to compare your experiments based on their parameters or their results.
+
+![](mlflow-ui.png)
+Figure 1: [A snapshot of the visual interface of MLflow](https://databricks.com/blog/2018/06/05/introducing-mlflow-an-open-source-machine-learning-platform.html).
+
+MLflow is the solution I use right now. It is minimal, easy to use and needs relatively low code (a couple of lines depending on the training framework you are using). 
```

Comment:
Pas certain de l'utilisation de _low_. Peut-être que je pense trop francophone.

Reply:
few lines of code?